streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamThread.java

```diff
@@ -705,7 +710,10 @@ private void runLoop() {
                         "This implies that this thread missed a rebalance and dropped out of the consumer group. " +
                         "Will try to rejoin the consumer group. Below is the detailed description of the task:\n{}",
                     ignoreAndRejoinGroup.migratedTask().id(), ignoreAndRejoinGroup.migratedTask().toString(">"));
-
+                enforceRebalance();
+            } catch (final ProducerFencedException | UnknownProducerIdException ignoreAndRejoinGroup) {
```

Comment:
Moving forward if we managed to make producer logic inside RecordCollector only then we do not need to capture those "raw" exceptions on the stream thread but only task-migrated exception.